### PR TITLE
[O2B-1119] Add QC Flag service

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- NA

Notable changes for developers:
- Added QC Flag service for flags fetching

Changes made to the database:
- Added sequelize migration file for QC Flag tables creation

### CODE REVIEW

File: lib/database/repositories/QcFlagRepository.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+const {
+    models: {
+        QcFlag,
+    },
+} = require('..');
```
Comment (martinboulais) [inline lib/database/repositories/QcFlagRepository.js:18]: One line

File: lib/database/seeders/20240404100811-qc-flags.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    up: async (queryInterface) =>
+        queryInterface.sequelize.transaction((transaction) =>
+            Promise.all([
+                queryInterface.bulkInsert('quality_control_flags', [
+
+                    /** Flags for runNumber: 106, LHC22b_apass1, CPV */
+                    {
+                        id: 1,
+                        from: new Date((1565314200 - 10000) * 1000),
+                        to: new Date((1565314200 + 10000) * 1000),
```
Comment (martinboulais) [inline lib/database/seeders/20240404100811-qc-flags.js:25]: Please use date like "2023-01-01T01:01:01", it will be way more clear when writing tests

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/database/repositories/QcFlagRepository.js (hunk 1 above)
Comment (xsalonx) [reply lib/database/repositories/QcFlagRepository.js]: done

File: lib/database/seeders/20240404100811-qc-flags.js (hunk 2 above)
Comment (xsalonx) [reply lib/database/seeders/20240404100811-qc-flags.js]: applied

**Review by martinboulais — APPROVED** [review]

**Pull request merged**